ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py

```diff
@@ -262,3 +262,43 @@ def test_corrcoef(
         y=x[1],
         rowvar=rowvar,
     )
+
+# bincount
+@handle_test(
+    fn_tree="functional.ivy.experimental.bincount",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=["int32", "int64"],
+        num_arrays=1,
+        shared_dtype=True,
+        min_num_dims=1,
+        max_num_dims=1,
+        min_dim_size=1,
+        max_dim_size=10,
+        min_value=0,
+        max_value=10,
+        allow_nan=False,
+    ),
+    weights=st.booleans(),
```

Comment:
Please generate the weights here instead of in the test, you can use st.one_of if you want to inlclude the case where it's `None`. You can use `key="shared_size"` in the above `helpers.dtype_and_values` function and whatever helper you use to generate the weights.